feat: add add how to contribute docs (#22)
dinngo/protocolink-logics: README.md

An SDK that build protocol logics for Protocolink

## How to contribute?

### Initial Setup

1. Fork this repository.

> 2. Set up your development environment and install the required dependencies.

Author: Could you provide more details about what needs to be added? The only thing that comes to mind initially is after forking the repository, just need to run `yarn`.

> 2. Set up your development environment and install the required dependencies.

Reviewer: There is `## Install` at the end of the README.

> 2. Set up your development environment and install the required dependencies.

Reviewer: Oh, the '## Install' section is intended for the client repository. I think adding `yarn` to the CLI section would make more sense.

> 2. Set up your development environment and install the required dependencies.

Author: Updated

### Contribution Steps

1. **Create a New Logic:**
  - Add a new folder under `src/logics` named after the protocol (protocolId), using Hyphenation format.
  - Create a new logic file under `src/logics/{protocolId}`, named with the action name (logicId). Prefix it with `logic.` and use Hyphenation format. The file name should be `logic.{logicId}.ts`.

2. **Implement the Logic:**
  - Implement the Logic class in the logic file using PascalCase. Append `Logic` to the action name.
  - Extend the `core.Logic` class.
  - Implement the following interfaces based on the requirements:

>     - `core.LogicTokenListInterface`: Support for tokens list function.

Reviewer 2: ```suggestion
    - `core.LogicTokenListInterface`: Support for tokens list functions.
```

Author: If there's only one function, should it be named in the plural form?

Author: Updated

>     - `core.LogicOracleInterface`: Quotation function.

Reviewer 2: ```suggestion
    - `core.LogicOracleInterface`: Quotation functions.
```

Author: If there's only one function, should it be named in the plural form?

Author: Updated

>     - `core.LogicBuilderInterface`: Txdata encode function for interacting with contracts.

Reviewer 2: ```suggestion
    - `core.LogicBuilderInterface`: Txdata encode functions for interacting with contracts.
```

Author: If there's only one function, should it be named in the plural form?

Author: Updated

  **Example:**
  ```typescript
  class LogicExampleLogic extends core.Logic implements
    core.LogicTokenListInterface, core.LogicOracleInterface, core.LogicBuilderInterface {
    // Your implementation here
  }
  ```

3. **Reference Examples:**
  - Explore reference examples for different categories:
    - Lending: [Aave V3](src/logics/aave-v2/), [Compound V3](src/logics/comopound-v3/)
    - Swap: [Uniswap V3](src/logics/uniswap-v3/), [ParaSwap V5](src/logics/paraswap-v5/)
    - FlashLoan: [Aave V3](src/logics/aave-v3/), [Balancer V2](src/logics/balancer-v2/)
  - More: [GO](src/logics/)

4. **Unit Testing:**

>   - Write tests that won't change based on block number increments.

Reviewer 2: ```suggestion
  - Write tests that won't fail based on block number increments.
```
Is this closer to the intended meaning?

Author: OK

Author: Updated

  - Test files are in the same path as the logic files, named `logic.{logicId}.test.ts`.
  - Write tests for functions like `getTokenList()` and `build()` as needed.

5. **Integration Testing:**
  - Write tests that interact with the Router contract in real time.

>   - Test files are in `test/logics/{protocolId}/{logicId}.test.ts`.

Reviewer 2: ```suggestion
  - Test files are in `test/logics/{protocolId}/`, named `{logicId}.test.ts`.
```

Author: Why?

Author: OK

Author: Updated

>   - Utilize Logic's quotation function to generate Logic Data for contract interaction.

Reviewer 2: ```suggestion
  - Utilize Logic's quotation functions to generate Logic Data for contract interactions.
```

Author: If there's only one function, should it be named in the plural form?

Author: Updated
